pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -309,6 +315,7 @@ func (tm *TableManager) ensureQueryReadiness(ctx context.Context) error {
 			return err
 		}
 
+		level.Debug(util_log.Logger).Log("msg", "instance should be query ready for users", "users", usersToBeQueryReadyFor, "query_readiness_duration", time.Since(start))
```

Comment:
I see 2 problems here:
1. We are logging the duration before doing query readiness.
2. We will be logging cumulative query readiness duration for tables since the `start` is never reset.

Reply:
ops, good catch. I've split it into two durations now: one per table, that I reset and only counts for `EnsureQueryReadiness`, and `start` is now only used to report the metric that reports the total time we spent here.